src/compiler/transformers/module/es6.ts

```diff
@@ -4,9 +4,35 @@
 /*@internal*/
 namespace ts {
     export function transformES6Module(context: TransformationContext) {
+        const compilerOptions = context.getCompilerOptions();
+        const resolver = context.getEmitResolver();
+
+        let currentSourceFile: SourceFile;
+
         return transformSourceFile;
 
         function transformSourceFile(node: SourceFile) {
+            if (isExternalModule(node) || compilerOptions.isolatedModules) {
+                currentSourceFile = node;
+                return visitEachChild(node, visitor, context);
+            }
+            return node;
+        }
+
+        function visitor(node: Node) {
+            switch (node.kind) {

```

Comment:
Seems a little overkill to use a switch - you could just turn this function into

```ts
return node.kind === SyntaxKind.ImportDeclaration ?
    visitImportDeclaration(node as  ImportDeclaration) :
    node;
```